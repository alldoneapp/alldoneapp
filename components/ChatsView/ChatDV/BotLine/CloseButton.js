import React from 'react'
import { StyleSheet } from 'react-native'

import Button from '../../../UIControls/Button'

export default function CloseButton({ setFullscreen }) {
    const closeFullscreen = () => {
        setFullscreen(false)
    }

    return (
        <Button
            ref={ref => (this.xBtnRef = ref)}
            type={'ghost'}
            noBorder={true}
            icon="x"
            onPress={closeFullscreen}
            buttonStyle={localStyles.container}
        />
    )
}

const localStyles = StyleSheet.create({
    container: {
        marginLeft: 8,
        marginRight: 20,
        padding: 10,
    },
})
