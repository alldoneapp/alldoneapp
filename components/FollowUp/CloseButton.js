import React, { useEffect } from 'react'
import { StyleSheet, TouchableOpacity, View } from 'react-native'

import { colors } from '../styles/global'
import Icon from '../Icon'

export default function CloseButton({ close, style, disabledEscape }) {
    const onPress = event => {
        setTimeout(() => {
            close(event)
        })
    }

    const onKeyDown = event => {
        const { key } = event

        if (key === 'Escape') {
            close(event)
        }
    }

    useEffect(() => {
        if (!disabledEscape) {
            document.addEventListener('keydown', onKeyDown)
            return () => {
                document.removeEventListener('keydown', onKeyDown)
            }
        }
    })

    return (
        <View style={[localStyles.closeContainer, style]}>
            <TouchableOpacity style={localStyles.closeButton} onPress={onPress}>
                <Icon name="x" size={24} color={colors.Text03} />
            </TouchableOpacity>
        </View>
    )
}

const localStyles = StyleSheet.create({
    closeContainer: {
        position: 'absolute',
        top: 13,
        right: 13,
    },
    closeButton: {
        alignItems: 'center',
        justifyContent: 'center',
    },
})
