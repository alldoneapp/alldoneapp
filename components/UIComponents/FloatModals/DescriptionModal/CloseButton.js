import React from 'react'
import { StyleSheet, TouchableOpacity, View } from 'react-native'
import Hotkeys from 'react-hot-keys'

import { colors } from '../../../styles/global'
import Icon from '../../../Icon'

export default function CloseButton({ closeModal }) {
    return (
        <View style={localStyles.closeContainer}>
            <Hotkeys keyName={'esc'} onKeyDown={(sht, event) => closeModal(event)} filter={e => true}>
                <TouchableOpacity
                    accessibilityLabel={'social-text-block'}
                    style={localStyles.closeButton}
                    onPress={() => closeModal()}
                >
                    <Icon accessibilityLabel={'social-text-block'} name="x" size={24} color={colors.Text03} />
                </TouchableOpacity>
            </Hotkeys>
        </View>
    )
}

const localStyles = StyleSheet.create({
    closeContainer: {
        position: 'absolute',
        top: 8,
        right: 8,
    },
    closeButton: {
        alignItems: 'center',
        justifyContent: 'center',
    },
})
