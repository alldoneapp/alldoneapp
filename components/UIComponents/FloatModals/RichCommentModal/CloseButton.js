import React, { useEffect } from 'react'
import { StyleSheet, TouchableOpacity, View } from 'react-native'

import { colors } from '../../../styles/global'
import Icon from '../../../Icon'

export default function CloseButton({ closeModal, comments }) {
    const onKeyDown = event => {
        const { key } = event

        if (key === 'Escape') {
            closeModal(event)
        }
    }

    useEffect(() => {
        document.addEventListener('keydown', onKeyDown)
        return () => {
            document.removeEventListener('keydown', onKeyDown)
        }
    })

    return (
        <View style={localStyles.closeContainer}>
            <TouchableOpacity
                disabled={!comments}
                accessibilityLabel={'social-text-block'}
                style={localStyles.closeButton}
                onPress={event => closeModal(comments && comments.length > 0 ? event : false)}
            >
                <Icon accessibilityLabel={'social-text-block'} name="x" size={24} color={colors.Text03} />
            </TouchableOpacity>
        </View>
    )
}

const localStyles = StyleSheet.create({
    closeContainer: {
        position: 'absolute',
        top: 0,
        right: 0,
    },
    closeButton: {
        alignItems: 'center',
        justifyContent: 'center',
    },
})
