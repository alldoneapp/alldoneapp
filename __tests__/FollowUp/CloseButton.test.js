import React from 'react'
import CloseButton from '../../components/FollowUp/CloseButton'
import renderer from 'react-test-renderer'

describe('CloseButton component', () => {
    describe('CloseButton snapshot test', () => {
        it('should render correctly', () => {
            const tree = renderer.create(<CloseButton close={() => {}} />).toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})
