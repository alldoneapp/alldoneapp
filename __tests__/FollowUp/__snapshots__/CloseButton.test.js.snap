// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton component CloseButton snapshot test should render correctly 1`] = `
<View
  style={
    Object {
      "position": "absolute",
      "right": 13,
      "top": 13,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#8A94A6",
            "fontFamily": "alldone",
            "fontSize": 24,
            "opacity": 1,
          }
        }
      >
        î¨¶
      </Text>
    </View>
  </View>
</View>
`;
